### PR TITLE
blobber slash from delegate balance

### LINKED ISSUE #2122: Blobber slash should be taken from stake pool not rewards.
[stakePool.Slash](https://github.com/0chain/0chain/blob/staging/code/go/0chain.net/smartcontract/storagesc/stakepool.go#L215)
```go
	if err := edbSlash.Emit(event.TagStakePoolReward, balances); err != nil {
		return 0, err
	}
```
This takes slash from rewards. This could result in negative rewards. The slash value would come from the stake pool instead. We need to modify event.DelegatePool to allow moidfy balances.

This code will be modified in both https://github.com/0chain/0chain/pull/1465 and [Blobber and validator, health check kill and shut down](https://github.com/0chain/0chain/pull/1439) so we can do this change as part of one of these or afterwards.

### PR DESCRIPTION
**Pull request by Hitenjain14**
## Fixes
Blobber slash is now deducted from [DelegatePool.Balance](https://github.com/0chain/0chain/blob/staging/code/go/0chain.net/smartcontract/dbs/event/delegate_pool.go#L24) and added to [DelegatePool.TotalPenalty](https://github.com/0chain/0chain/blob/staging/code/go/0chain.net/smartcontract/dbs/event/delegate_pool.go#L27).
#2122 

## Changes
Added new tag for `TagStakePoolPenalty`

Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/0chain/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

## Associated PRs (Link as appropriate):
- blobber:
- gosdk:
- system_test:
- zboxcli:
- zwalletcli:
- Other: ...

### CODE REVIEW

File: code/go/0chain.net/smartcontract/storagesc/stakepool.go [hunk 1]
```diff
@@ -259,8 +259,8 @@ func (sp *stakePool) slash(
 		}
 		edbSlash.DelegatePenalties[id] = dpSlash
 	}
-	// todo we should slash from stake pools not rewards. 0chain issue 1495
-	if err := edbSlash.Emit(event.TagStakePoolReward, balances); err != nil {
+	//Added New Tag for StakePoolPenalty
+	if err := edbSlash.Emit(event.TagStakePoolPenalty, balances); err != nil {
```
Comment (dabasov) [inline code/go/0chain.net/smartcontract/storagesc/stakepool.go:263]: is stake pool reduced for this amount? I don't see the reduction. At first we should reduce the stakepool and all the delegate pools proportionally in the MPT, and only after it update the events_db. looks like with current implementation events_db will be changed, but MPT won't and balances will be different in different storages

**Review by dabasov — CHANGES_REQUESTED** [review]

File: code/go/0chain.net/smartcontract/storagesc/stakepool.go (hunk 1 above)
Comment (dabasov) [reply code/go/0chain.net/smartcontract/storagesc/stakepool.go:263]: here is the code 
```go
		if balance, err := currency.MinusCoin(dp.Balance, dpSlash); err != nil {
			return 0, err
		} else {
			dp.Balance = balance
		}
```

**Review by dabasov — APPROVED** [review]
lgtm

**Comment by Kishan-Dhakan** [comment]
@Hitenjain14 can we add system-tests for this?

**Review by peterlimg — APPROVED** [review]
LGTM

**Pull request merged**